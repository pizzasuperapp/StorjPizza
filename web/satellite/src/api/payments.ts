// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { ErrorUnauthorized } from '@/api/errors/ErrorUnauthorized';
import {
    AccountBalance,
    Coupon,
    CreditCard,
    PaymentsApi,
    PaymentsHistoryItem,
    ProjectUsageAndCharges,
    TokenDeposit
} from '@/types/payments';
import { HttpClient } from '@/utils/httpClient';
import { Time } from '@/utils/time';
import { ErrorTooManyRequests } from './errors/ErrorTooManyRequests';

/**
 * PaymentsHttpApi is a http implementation of Payments API.
 * Exposes all payments-related functionality
 */
export class PaymentsHttpApi implements PaymentsApi {
    private readonly client: HttpClient = new HttpClient();
    private readonly ROOT_PATH: string = '/api/v0/payments';

    /**
     * Get account balance.
     *
     * @returns balance in cents
     * @throws Error
     */
    public async getBalance(): Promise<AccountBalance> {
        const path = `${this.ROOT_PATH}/account/balance`;
        const response = await this.client.get(path);

        if (!response.ok) {
            if (response.status === 401) {
                throw new ErrorUnauthorized();
            }

            throw new Error('Can not get account balance');
        }

        const balance = await response.json();
        if (balance) {
            return new AccountBalance(balance.freeCredits, balance.coins);
        }

        return new AccountBalance();
    }

    /**
     * Try to set up a payment account.
     *
     * @throws Error
     */
    public async setupAccount(): Promise<string> {
        const path = `${this.ROOT_PATH}/account`;
        const response = await this.client.post(path, null);
        const couponType = await response.json();

        if (response.ok) {
            return couponType;
        }

        if (response.status === 401) {
            throw new ErrorUnauthorized();
        }

        throw new Error('can not setup account');
    }

    /**
     * projectsUsageAndCharges returns usage and how much money current user will be charged for each project which he owns.
     */
    public async projectsUsageAndCharges(start: Date, end: Date): Promise<ProjectUsageAndCharges[]> {
        const since = Time.toUnixTimestamp(start).toString();
        const before = Time.toUnixTimestamp(end).toString();
        const path = `${this.ROOT_PATH}/account/charges?from=${since}&to=${before}`;
        const response = await this.client.get(path);

        if (!response.ok) {
            if (response.status === 401) {
                throw new ErrorUnauthorized();
            }

            throw new Error('can not get projects charges');
        }

        const charges = await response.json();
        if (charges) {
            return charges.map(charge =>
                new ProjectUsageAndCharges(
                    new Date(charge.since),
                    new Date(charge.before),
                    charge.egress,
                    charge.storage,
                    charge.segmentCount,
                    charge.projectId,
                    charge.storagePrice,
                    charge.egressPrice,
                    charge.segmentPrice,
                ),
            );
        }

        return [];
    }

    /**
     * Add credit card.
     *
     * @param token - stripe token used to add a credit card as a payment method
     * @throws Error
     */
    public async addCreditCard(token: string): Promise<void> {
        const path = `${this.ROOT_PATH}/cards`;
        const response = await this.client.post(path, token);

        if (response.ok) {
            return;
        }

        if (response.status === 401) {
            throw new ErrorUnauthorized();
        }

        throw new Error('can not add credit card');
    }

    /**
     * Detach credit card from payment account.
     *
     * @param cardId
     * @throws Error
     */
    public async removeCreditCard(cardId: string): Promise<void> {
        const path = `${this.ROOT_PATH}/cards/${cardId}`;
        const response = await this.client.delete(path);

        if (response.ok) {
            return;
        }

        if (response.status === 401) {
            throw new ErrorUnauthorized();
        }

        throw new Error('can not remove credit card');
    }

    /**
     * Get list of user`s credit cards.
     *
     * @returns list of credit cards
     * @throws Error
     */
    public async listCreditCards(): Promise<CreditCard[]> {
        const path = `${this.ROOT_PATH}/cards`;
        const response = await this.client.get(path);

        if (!response.ok) {
            if (response.status === 401) {
                throw new ErrorUnauthorized();
            }
            throw new Error('can not list credit cards');
        }

        const creditCards = await response.json();

        if (creditCards) {
            return creditCards.map(card => new CreditCard(card.id, card.expMonth, card.expYear, card.brand, card.last4, card.isDefault));
        }

        return [];
    }

    /**
     * Make credit card default.
     *
     * @param cardId
     * @throws Error
     */
    public async makeCreditCardDefault(cardId: string): Promise<void> {
        const path = `${this.ROOT_PATH}/cards`;
        const response = await this.client.patch(path, cardId);

        if (response.ok) {
            return;
        }

        if (response.status === 401) {
            throw new ErrorUnauthorized();
        }

        throw new Error('can not make credit card default');
    }

    /**
     * Returns a list of invoices, transactions and all others payments history items for payment account.
     *
     * @returns list of payments history items
     * @throws Error
     */
    public async paymentsHistory(): Promise<PaymentsHistoryItem[]> {
        const path = `${this.ROOT_PATH}/billing-history`;
        const response = await this.client.get(path);

        if (!response.ok) {
            if (response.status === 401) {
                throw new ErrorUnauthorized();
            }
            throw new Error('can not list billing history');
        }

        const paymentsHistoryItems = await response.json();
        if (paymentsHistoryItems) {
            return paymentsHistoryItems.map(item =>
                new PaymentsHistoryItem(
                    item.id,
                    item.description,
                    item.amount,
                    item.received,
                    item.status,
                    item.link,
                    new Date(item.start),
                    new Date(item.end),
                    item.type,
                    item.remaining,
                ),
            );
        }

        return [];
    }

    /**
     * makeTokenDeposit process coin payments.
     *
     * @param amount
     * @throws Error
     */
    public async makeTokenDeposit(amount: number): Promise<TokenDeposit> {
        const path = `${this.ROOT_PATH}/tokens/deposit`;
        const response = await this.client.post(path, JSON.stringify({ amount }));

        if (!response.ok) {
            if (response.status === 401) {
                throw new ErrorUnauthorized();
            }

            throw new Error('can not process coin payment');
        }

        const result = await response.json();

        return new TokenDeposit(result.amount, result.address, result.link);
    }

    /**
     * applyCouponCode applies a coupon code.
     *
     * @param couponCode
     * @throws Error
     */
    public async applyCouponCode(couponCode: string): Promise<Coupon> {
        const path = `${this.ROOT_PATH}/coupon/apply`;
        const response = await this.client.patch(path, couponCode);
        const errMsg = `Could not apply coupon code "${couponCode}"`;

        if (response.ok) {
            const coupon = await response.json();

            if (!coupon) {
                throw new Error(errMsg);
            }

            return new Coupon(
                coupon.id,
                coupon.promoCode,
                coupon.name,
                coupon.amountOff,
                coupon.percentOff,
                new Date(coupon.addedAt),
                coupon.expiresAt ? new Date(coupon.expiresAt) : null,
                coupon.duration
            );
        }

        switch (response.status) {
        case 429:
            throw new ErrorTooManyRequests('You\'ve exceeded limit of attempts, try again in 5 minutes');
        case 401:
            throw new ErrorUnauthorized(errMsg);
        default:
            throw new Error(errMsg);
        }
    }

    /**
     * getCoupon returns the coupon applied to the user.
     *
     * @throws Error
     */
    public async getCoupon(): Promise<Coupon | null> {
        const path = `${this.ROOT_PATH}/coupon`;
        const response = await this.client.get(path);
        if (!response.ok) {
            if (response.status === 401) {
                throw new ErrorUnauthorized();
            }

            throw new Error('cannot retrieve coupon');
        }

        const coupon = await response.json();

        if (!coupon) {
            return null;
        }

        return new Coupon(
            coupon.id,
            coupon.promoCode,
            coupon.name,
            coupon.amountOff,
            coupon.percentOff,
            new Date(coupon.addedAt),
            coupon.expiresAt ? new Date(coupon.expiresAt) : null,
            coupon.duration
        );
    }
}
