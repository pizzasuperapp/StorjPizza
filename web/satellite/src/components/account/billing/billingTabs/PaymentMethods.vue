// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div>
        Payment Methods
        <router-view />
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

// @vue/component
@Component({
    components: {
    },
})
export default class BillingArea extends Vue {

}
</script>