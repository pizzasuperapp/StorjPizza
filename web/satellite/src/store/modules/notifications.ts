// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { DelayedNotification } from '@/types/DelayedNotification';
import { NOTIFICATION_ACTIONS } from '@/utils/constants/actionNames';
import { NOTIFICATION_TYPES } from '@/utils/constants/notification';

import { NOTIFICATION_MUTATIONS } from '../mutationConstants';
import {StoreModule} from "@/types/store";

export class NotificationsState {
    public notificationQueue: DelayedNotification[] = [];
}

interface NotificationsContext {
    state: NotificationsState
    commit: (string, ...unknown) => void
}

export function makeNotificationsModule(): StoreModule<NotificationsState, NotificationsContext> {
    return {
        state: new NotificationsState(),
        mutations: {
            // Mutation for adding notification to queue
            [NOTIFICATION_MUTATIONS.ADD](state: NotificationsState, notification: DelayedNotification): void {
                state.notificationQueue.push(notification);
            },
            // Mutation for deleting notification to queue
            [NOTIFICATION_MUTATIONS.DELETE](state: NotificationsState, id: string): void {
                if (state.notificationQueue.length < 1) {
                    return;
                }

                const selectedNotification =  getNotificationById(state.notificationQueue, id);
                if (selectedNotification) {
                    selectedNotification.pause();
                    state.notificationQueue.splice(state.notificationQueue.indexOf(selectedNotification), 1);
                }
            },
            [NOTIFICATION_MUTATIONS.PAUSE](state: NotificationsState, id: string): void {
                const selectedNotification =  getNotificationById(state.notificationQueue, id);
                if (selectedNotification) {
                    selectedNotification.pause();
                }
            },
            [NOTIFICATION_MUTATIONS.RESUME](state: NotificationsState, id: string): void {
                const selectedNotification =  getNotificationById(state.notificationQueue, id);
                if (selectedNotification) {
                    selectedNotification.start();
                }
            },
            [NOTIFICATION_MUTATIONS.CLEAR](state: NotificationsState): void {
                state.notificationQueue = [];
            },
        },
        actions: {
            // Commits mutation for adding success notification
            [NOTIFICATION_ACTIONS.SUCCESS]: function ({commit}: NotificationsContext, message: string): void {
                const notification = new DelayedNotification(
                    () => commit(NOTIFICATION_MUTATIONS.DELETE, notification.id),
                    NOTIFICATION_TYPES.SUCCESS,
                    message,
                );

                commit(NOTIFICATION_MUTATIONS.ADD, notification);
            },
            // Commits mutation for adding info notification
            [NOTIFICATION_ACTIONS.NOTIFY]: function ({commit}: NotificationsContext, message: string): void {

                const notification = new DelayedNotification(
                    () => commit(NOTIFICATION_MUTATIONS.DELETE, notification.id),
                    NOTIFICATION_TYPES.NOTIFICATION,
                    message,
                );

                commit(NOTIFICATION_MUTATIONS.ADD, notification);
            },
            // Commits mutation for adding error notification
            [NOTIFICATION_ACTIONS.ERROR]: function ({commit}: NotificationsContext, message: string): void {
                const notification = new DelayedNotification(
                    () => commit(NOTIFICATION_MUTATIONS.DELETE, notification.id),
                    NOTIFICATION_TYPES.ERROR,
                    message,
                );

                commit(NOTIFICATION_MUTATIONS.ADD, notification);
            },
            // Commits mutation for adding error notification
            [NOTIFICATION_ACTIONS.WARNING]: function ({commit}: NotificationsContext, message: string): void {
                const notification = new DelayedNotification(
                    () => commit(NOTIFICATION_MUTATIONS.DELETE, notification.id),
                    NOTIFICATION_TYPES.WARNING,
                    message,
                );

                commit(NOTIFICATION_MUTATIONS.ADD, notification);
            },
            [NOTIFICATION_ACTIONS.DELETE]: function ({commit}: NotificationsContext, id: string): void {
                commit(NOTIFICATION_MUTATIONS.DELETE, id);
            },
            [NOTIFICATION_ACTIONS.PAUSE]: function ({commit}: NotificationsContext, id: string): void {
                commit(NOTIFICATION_MUTATIONS.PAUSE, id);
            },
            [NOTIFICATION_ACTIONS.RESUME]: function ({commit}: NotificationsContext, id: string): void {
                commit(NOTIFICATION_MUTATIONS.RESUME, id);
            },
            [NOTIFICATION_ACTIONS.CLEAR]: function ({commit}): void {
                commit(NOTIFICATION_MUTATIONS.CLEAR);
            },
        },
        getters: {},
    };
}

function getNotificationById(notifications: DelayedNotification[], id: string): DelayedNotification | undefined {
    return notifications.find((notification: DelayedNotification) => notification.id === id);
}
