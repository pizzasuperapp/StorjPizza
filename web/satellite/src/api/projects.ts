// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { BaseGql } from '@/api/baseGql';
import { ErrorUnauthorized } from '@/api/errors/ErrorUnauthorized';
import {
    DataStamp,
    Project,
    ProjectFields,
    ProjectLimits,
    ProjectsApi,
    ProjectsCursor,
    ProjectsPage,
    ProjectsStorageBandwidthDaily,
} from '@/types/projects';
import { HttpClient } from '@/utils/httpClient';
import { Time } from "@/utils/time";

export class ProjectsApiGql extends BaseGql implements ProjectsApi {
    private readonly http: HttpClient = new HttpClient();
    private readonly ROOT_PATH: string = '/api/v0/projects';

    /**
     * Creates project.
     *
     * @param projectFields - contains project information
     * @throws Error
     */
    public async create(projectFields: ProjectFields): Promise<Project> {
        const query =
            `mutation($name: String!, $description: String!) {
                createProject(
                    input: {
                        name: $name,
                        description: $description,
                    }
                ) {id}
            }`;

        const variables = {
            name: projectFields.name,
            description: projectFields.description,
        };

        const response = await this.mutate(query, variables);

        return new Project(response.data.createProject.id, variables.name, variables.description, '', projectFields.ownerId);
    }

    /**
     * Fetch projects.
     *
     * @returns Project[]
     * @throws Error
     */
    public async get(): Promise<Project[]> {
        const query = `query {
            myProjects{
                name
                id
                description
                createdAt
                ownerId
            }
        }`;

        const response = await this.query(query);

        return response.data.myProjects.map((project: Project) => {
            return new Project(
                project.id,
                project.name,
                project.description,
                project.createdAt,
                project.ownerId,
            );
        });
    }

    /**
     * Update project name and description.
     *
     * @param projectId - project ID
     * @param projectFields - project fields
     * @param projectLimits - project limits
     * @returns Project[]
     * @throws Error
     */
    public async update(projectId: string, projectFields: ProjectFields, projectLimits: ProjectLimits): Promise<void> {
        const query =
            `mutation($projectId: String!, $name: String!, $description: String!, $storageLimit: String!, $bandwidthLimit: String!) {
                updateProject(
                    id: $projectId,
                    projectFields: {
                        name: $name,
                        description: $description,
                    },
                    projectLimits: {
                        storageLimit: $storageLimit,
                        bandwidthLimit: $bandwidthLimit,
                    }
                ) {name}
            }`;

        const variables = {
            projectId: projectId,
            name: projectFields.name,
            description: projectFields.description,
            storageLimit: projectLimits.storageLimit.toString(),
            bandwidthLimit: projectLimits.bandwidthLimit.toString(),
        };

        await this.mutate(query, variables);
    }

    /**
     * Delete project.
     *
     * @param projectId - project ID
     * @throws Error
     */
    public async delete(projectId: string): Promise<void> {
        const query =
            `mutation($projectId: String!) {
                deleteProject(
                    id: $projectId
                ) {name}
            }`;

        const variables = {
            projectId: projectId,
        };

        await this.mutate(query, variables);
    }

    /**
     * Get project limits.
     *
     * @param projectId- project ID
     * throws Error
     */
    public async getLimits(projectId: string): Promise<ProjectLimits> {
        const path = `${this.ROOT_PATH}/${projectId}/usage-limits`;
        const response = await this.http.get(path);

        if (response.ok) {
            const limits = await response.json();

            return new ProjectLimits(
                limits.bandwidthLimit,
                limits.bandwidthUsed,
                limits.storageLimit,
                limits.storageUsed,
                limits.objectCount,
                limits.segmentCount,
            );
        }

        if (response.status === 401) {
            throw new ErrorUnauthorized();
        }

        throw new Error('can not get usage limits');
    }

    /**
     * Get total limits for all the projects that user owns.
     *
     * throws Error
     */
    public async getTotalLimits(): Promise<ProjectLimits> {
        const path = `${this.ROOT_PATH}/usage-limits`;
        const response = await this.http.get(path);

        if (response.ok) {
            const limits = await response.json();

            return new ProjectLimits(
                limits.bandwidthLimit,
                limits.bandwidthUsed,
                limits.storageLimit,
                limits.storageUsed,
            );
        }

        if (response.status === 401) {
            throw new ErrorUnauthorized();
        }

        throw new Error('can not get total usage limits');
    }

    /**
     * Get project daily usage for specific date range.
     *
     * @param projectId- project ID
     * @param start- since date
     * @param end- before date
     * throws Error
     */
    public async getDailyUsage(projectId: string, start: Date, end: Date): Promise<ProjectsStorageBandwidthDaily> {
        const since = Time.toUnixTimestamp(start).toString();
        const before = Time.toUnixTimestamp(end).toString();
        const path = `${this.ROOT_PATH}/${projectId}/daily-usage?from=${since}&to=${before}`;
        const response = await this.http.get(path);

        if (response.ok) {
            const usage = await response.json();

            return new ProjectsStorageBandwidthDaily(
                usage.storageUsage.map(el => {
                    const date = new Date(el.date)
                    date.setHours(0, 0, 0, 0)
                    return new DataStamp(el.value, date)
                }),
                usage.allocatedBandwidthUsage.map(el => {
                    const date = new Date(el.date)
                    date.setHours(0, 0, 0, 0)
                    return new DataStamp(el.value, date)
                }),
                usage.settledBandwidthUsage.map(el => {
                    const date = new Date(el.date)
                    date.setHours(0, 0, 0, 0)
                    return new DataStamp(el.value, date)
                }),
            );
        }

        if (response.status === 401) {
            throw new ErrorUnauthorized();
        }

        throw new Error('can not get project daily usage');
    }

    /**
     * Fetch owned projects.
     *
     * @returns ProjectsPage
     * @throws Error
     */
    public async getOwnedProjects(cursor: ProjectsCursor): Promise<ProjectsPage> {
        const query =
            `query($limit: Int!, $page: Int!) {
                ownedProjects( cursor: { limit: $limit, page: $page } ) {
                    projects {
                        id,
                        name,
                        ownerId,
                        description,
                        createdAt,
                        memberCount
                    },
                    limit,
                    offset,
                    pageCount,
                    currentPage,
                    totalCount
                 }
             }`;

        const variables = {
            limit: cursor.limit,
            page: cursor.page,
        };

        const response = await this.query(query, variables);

        return this.getProjectsPage(response.data.ownedProjects);
    }

    /**
     * Method for mapping projects page from json to ProjectsPage type.
     *
     * @param page anonymous object from json
     */
    private getProjectsPage(page: any): ProjectsPage { // eslint-disable-line @typescript-eslint/no-explicit-any
        if (!page) {
            return new ProjectsPage();
        }

        const projects: Project[] = page.projects.map(key =>
            new Project(
                key.id,
                key.name,
                key.description,
                key.createdAt,
                key.ownerId,
                false,
                key.memberCount));

        return new ProjectsPage(projects, page.limit, page.offset, page.pageCount, page.currentPage, page.totalCount);
    }

}
