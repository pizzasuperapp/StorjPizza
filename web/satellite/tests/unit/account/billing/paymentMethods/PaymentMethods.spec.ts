// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import Vuex from 'vuex';

import PaymentMethods from '@/components/account/billing/paymentMethods/PaymentMethods.vue';

import { appStateModule } from '@/store/modules/appState';
import { makeNotificationsModule } from '@/store/modules/notifications';
import { makePaymentsModule, PAYMENTS_MUTATIONS } from '@/store/modules/payments';
import { makeProjectsModule } from '@/store/modules/projects';
import { makeUsersModule, USER_MUTATIONS } from '@/store/modules/users';
import { CreditCard } from '@/types/payments';
import { User } from '@/types/users';
import { NotificatorPlugin } from '@/utils/plugins/notificator';
import { createLocalVue, mount, shallowMount } from '@vue/test-utils';

import { PaymentsMock } from '../../../mock/api/payments';
import { ProjectsApiMock } from '../../../mock/api/projects';
import { UsersApiMock } from '../../../mock/api/users';

const localVue = createLocalVue();

localVue.use(Vuex);

const paymentsApi = new PaymentsMock();
const paymentsModule = makePaymentsModule(paymentsApi);
const usersApi = new UsersApiMock();
const usersModule = makeUsersModule(usersApi);
const projectsApi = new ProjectsApiMock();
const projectsModule = makeProjectsModule(projectsApi);
const notificationsModule = makeNotificationsModule();
const store = new Vuex.Store({ modules: { usersModule, paymentsModule, projectsModule, appStateModule, notificationsModule }});
store.commit(USER_MUTATIONS.SET_USER, new User('id', 'name', 'short', 'test@test.test', 'partner', 'pass'));

localVue.use(new NotificatorPlugin(store));

const ANIMATION_COMPLETE_TIME = 600;

describe('PaymentMethods', () => {
    it('renders correctly without card', async (): Promise<void> => {
        const wrapper = mount(PaymentMethods, {
            store,
            localVue,
        });

        await wrapper.setData({ areCardsFetching: false });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly after add card and cancel click', async () => {
        const wrapper = shallowMount(PaymentMethods, {
            store,
            localVue,
        });

        await wrapper.setData({ areCardsFetching: false });

        await wrapper.find('.add-card-button').trigger('click');

        await new Promise<void>(resolve => {
            setTimeout(() => {
                expect(wrapper).toMatchSnapshot();
                wrapper.find('.payment-methods-area__functional-area__button-area__cancel').trigger('click');

                setTimeout(() => {
                    expect(wrapper).toMatchSnapshot();
                    resolve();
                }, ANIMATION_COMPLETE_TIME);
            }, ANIMATION_COMPLETE_TIME);
        });
    });

    it('renders correctly after add STORJ and cancel click', async () => {
        const wrapper = mount(PaymentMethods, {
            store,
            localVue,
        });

        await wrapper.setData({ areCardsFetching: false });

        await wrapper.find('.add-storj-button').trigger('click');

        await new Promise<void>(resolve => {
            setTimeout(() => {
                expect(wrapper).toMatchSnapshot();
                wrapper.find('.payment-methods-area__functional-area__button-area__cancel').trigger('click');

                setTimeout(() => {
                    expect(wrapper).toMatchSnapshot();
                    resolve();
                }, ANIMATION_COMPLETE_TIME);
            }, ANIMATION_COMPLETE_TIME);
        });
    });

    it('renders correctly with card', async (): Promise<void> => {
        const card = new CreditCard('cardId', 12, 2100, 'test', '0000', true);
        store.commit(PAYMENTS_MUTATIONS.SET_CREDIT_CARDS, [card]);

        const wrapper = mount(PaymentMethods, {
            store,
            localVue,
        });

        await wrapper.setData({ areCardsFetching: false });

        expect(wrapper).toMatchSnapshot();
    });
});