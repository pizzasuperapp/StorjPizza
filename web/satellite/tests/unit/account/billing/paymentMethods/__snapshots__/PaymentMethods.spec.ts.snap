// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethods renders correctly after add STORJ and cancel click 1`] = `
<div class="payment-methods-area">
  <div class="payment-methods-area__functional-area extended">
    <div class="payment-methods-area__functional-area__top-container">
      <h1 aria-roledescription="title" class="payment-methods-area__functional-area__title">Payment Method</h1>
      <div class="payment-methods-area__functional-area__button-area">
        <div class="payment-methods-area__functional-area__button-area__cancel">
          <p class="payment-methods-area__functional-area__button-area__cancel__text">Cancel</p>
        </div>
      </div>
    </div>
    <div class="payments-bonus-container payment-methods-area__functional-area__bonus"><svg class="banner-logo-svg"></svg>
      <div class="payments-bonus-container__text-container">
        <p class="payments-bonus-container__text-container__main-text">
          Get free credits for paying in STORJ Token
        </p>
        <p class="payments-bonus-container__text-container__additional-text">
          Deposit STORJ Token to your account and receive a 10% bonus, or $10 for every $100.
        </p>
      </div>
    </div>
    <div class="add-storj-area">
      <div class="add-storj-area__selection-container">
        <p class="add-storj-area__selection-container__label">Deposit STORJ Tokens via Coin Payments</p>
        <div class="form-container add-storj-area__selection-container__form">
          <div class="selected-container">
            <div class="selected-container__label-container">
              <p class="selected-container__label-container__label">USD $10</p>
              <div class="selected-container__label-container__svg"><svg width="14" height="8" viewBox="0 0 14 8" fill="none" xmlns="http://www.w3.org/2000/svg">
                  <path fill-rule="evenodd" clip-rule="evenodd" d="M0.372773 0.338888C0.869804 -0.112963 1.67565 -0.112963 2.17268 0.338888L7 4.72741L11.8273 0.338888C12.3243 -0.112963 13.1302 -0.112963 13.6272 0.338888C14.1243 0.790739 14.1243 1.52333 13.6272 1.97519L7 8L0.372773 1.97519C-0.124258 1.52333 -0.124258 0.790739 0.372773 0.338888Z" fill="#2683FF"></path>
                </svg></div>
            </div>
          </div>
          <!---->
          <!---->
          <!---->
        </div>
      </div>
      <div class="add-storj-area__submit-area">
        <!---->
        <div class="confirm-add-storj-button container" style="width: 251px; height: 48px; border-radius: 6px; font-size: 16px;">
          <!---->
          <!---->
          <!----> <span class="label">Continue to Coin Payments</span>
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
</div>
`;

exports[`PaymentMethods renders correctly after add STORJ and cancel click 2`] = `
<div class="payment-methods-area">
  <div class="payment-methods-area__functional-area">
    <div class="payment-methods-area__functional-area__top-container">
      <h1 aria-roledescription="title" class="payment-methods-area__functional-area__title">Payment Method</h1>
      <div class="payment-methods-area__functional-area__button-area">
        <div class="payment-methods-area__functional-area__button-area__default-buttons">
          <div class="add-storj-button container blue-white" style="width: 123px; height: 46px; border-radius: 6px; font-size: 16px;">
            <!---->
            <!---->
            <!----> <span class="label">Add STORJ</span>
          </div>
        </div>
      </div>
    </div>
    <div class="payments-bonus-container payment-methods-area__functional-area__bonus"><svg class="banner-gift-svg"></svg>
      <div class="payments-bonus-container__text-container">
        <p class="payments-bonus-container__text-container__main-text">
          Add a Payment Method to Get Started
        </p>
        <p class="payments-bonus-container__text-container__additional-text">
          Your first 150GB are free. You will not be charged until you use more than 150GB.
        </p>
      </div>
    </div>
    <!---->
    <!---->
    <div class="add-card-button">
      <!---->
      <!----> <span>Add Card</span>
    </div>
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
</div>
`;

exports[`PaymentMethods renders correctly after add card and cancel click 1`] = `
<div class="payment-methods-area">
  <div class="payment-methods-area__functional-area reduced">
    <div class="payment-methods-area__functional-area__top-container">
      <h1 aria-roledescription="title" class="payment-methods-area__functional-area__title">Payment Method</h1>
      <div class="payment-methods-area__functional-area__button-area">
        <div class="payment-methods-area__functional-area__button-area__cancel">
          <p class="payment-methods-area__functional-area__button-area__cancel__text">Cancel</p>
        </div>
      </div>
    </div>
    <!---->
    <!---->
    <addcardform-stub></addcardform-stub>
    <div class="add-card-button button-moved">
      <!---->
      <!----> <span>Add Card</span>
    </div>
  </div>
  <div class="payment-methods-area__security-info-container">
    <lockimage-stub></lockimage-stub> <span class="payment-methods-area__security-info-container__info">
            Your card is secured by Stripe through TLS and AES-256 encryption. Your information is secure.
        </span>
  </div>
  <!---->
  <!---->
  <!---->
</div>
`;

exports[`PaymentMethods renders correctly after add card and cancel click 2`] = `
<div class="payment-methods-area">
  <div class="payment-methods-area__functional-area">
    <div class="payment-methods-area__functional-area__top-container">
      <h1 aria-roledescription="title" class="payment-methods-area__functional-area__title">Payment Method</h1>
      <div class="payment-methods-area__functional-area__button-area">
        <div class="payment-methods-area__functional-area__button-area__default-buttons">
          <vbutton-stub label="Add STORJ" width="123px" height="46px" fontsize="16px" borderradius="6px" isbluewhite="true" onpress="[Function]" class="add-storj-button"></vbutton-stub>
        </div>
      </div>
    </div>
    <paymentsbonus-stub class="payment-methods-area__functional-area__bonus"></paymentsbonus-stub>
    <!---->
    <!---->
    <div class="add-card-button">
      <!---->
      <!----> <span>Add Card</span>
    </div>
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
</div>
`;

exports[`PaymentMethods renders correctly with card 1`] = `
<div class="payment-methods-area">
  <div class="payment-methods-area__functional-area">
    <div class="payment-methods-area__functional-area__top-container">
      <h1 aria-roledescription="title" class="payment-methods-area__functional-area__title">Payment Method</h1>
      <div class="payment-methods-area__functional-area__button-area">
        <div class="payment-methods-area__functional-area__button-area__default-buttons">
          <div class="add-storj-button container blue-white" style="width: 123px; height: 46px; border-radius: 6px; font-size: 16px;">
            <!---->
            <!---->
            <!----> <span class="label">Add STORJ</span>
          </div>
        </div>
      </div>
    </div>
    <div class="payments-bonus-container payment-methods-area__functional-area__bonus"><svg class="banner-logo-svg"></svg>
      <div class="payments-bonus-container__text-container">
        <p class="payments-bonus-container__text-container__main-text">
          Get free credits for paying in STORJ Token
        </p>
        <p class="payments-bonus-container__text-container__additional-text">
          Deposit STORJ Token to your account and receive a 10% bonus, or $10 for every $100.
        </p>
      </div>
    </div>
    <!---->
    <!---->
    <div class="add-card-button">
      <!---->
      <!----> <span>Add Card</span>
    </div>
  </div>
  <!---->
  <!---->
  <div class="payment-methods-area__existing-cards-container">
    <div class="payment-methods-container__card-container">
      <div class="payment-methods-container__card-container__info-area">
        <div class="payment-methods-container__card-container__info-area__card-logo"><svg></svg></div>
        <div class="payment-methods-container__card-container__info-area__info-container"><img src="@/../static/images/payments/cardStars.png" alt="Hidden card digits stars image">
          <h1 class="bold">0000</h1>
        </div>
        <div class="payment-methods-container__card-container__info-area__expire-container">
          <h2 class="medium">Expires</h2>
          <h1 class="bold">12/2100</h1>
        </div>
      </div>
      <div class="payment-methods-container__card-container__button-area">
        <div class="payment-methods-container__card-container__default-button">
          <p class="payment-methods-container__card-container__default-button__label">Default</p>
        </div>
      </div>
    </div>
  </div>
  <!---->
</div>
`;

exports[`PaymentMethods renders correctly without card 1`] = `
<div class="payment-methods-area">
  <div class="payment-methods-area__functional-area">
    <div class="payment-methods-area__functional-area__top-container">
      <h1 aria-roledescription="title" class="payment-methods-area__functional-area__title">Payment Method</h1>
      <div class="payment-methods-area__functional-area__button-area">
        <div class="payment-methods-area__functional-area__button-area__default-buttons">
          <div class="add-storj-button container blue-white" style="width: 123px; height: 46px; border-radius: 6px; font-size: 16px;">
            <!---->
            <!---->
            <!----> <span class="label">Add STORJ</span>
          </div>
        </div>
      </div>
    </div>
    <div class="payments-bonus-container payment-methods-area__functional-area__bonus"><svg class="banner-gift-svg"></svg>
      <div class="payments-bonus-container__text-container">
        <p class="payments-bonus-container__text-container__main-text">
          Add a Payment Method to Get Started
        </p>
        <p class="payments-bonus-container__text-container__additional-text">
          Your first 150GB are free. You will not be charged until you use more than 150GB.
        </p>
      </div>
    </div>
    <!---->
    <!---->
    <div class="add-card-button">
      <!---->
      <!----> <span>Add Card</span>
    </div>
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
</div>
`;
